Sources/Build/BuildDescription/ProductBuildDescription.swift

```diff
@@ -267,16 +267,17 @@ public final class ProductBuildDescription: SPMBuildCore.ProductBuildDescription
 
             // When deploying to macOS prior to macOS 12, add an rpath to the
             // back-deployed concurrency libraries.
-            if useStdlibRpath, self.buildParameters.targetTriple.isMacOSX,
-               let macOSSupportedPlatform = self.package.platforms.getDerived(for: .macOS),
-               macOSSupportedPlatform.version.major < 12
+            if useStdlibRpath, self.buildParameters.targetTriple.isMacOSX
             {
```

Comment:
another formatting nit
```suggestion
            if useStdlibRpath, self.buildParameters.targetTriple.isMacOSX {
```